qiskit/algorithms/minimum_eigen_solvers/vqe.py

```diff
@@ -318,7 +324,7 @@ def construct_expectation(
 
         observable_meas = self.expectation.convert(StateFn(operator, is_measurement=True))
         ansatz_circuit_op = CircuitStateFn(wave_function)
-        return observable_meas.compose(ansatz_circuit_op).reduce()
```

Comment:
Why is the call to `reduce` removed? 🙂

Reply:
This remove copies ansatz's quantum circuit. In order to take advantage of lazy evaluation, we should reduce at a later stage. Since CircuitSampler do the `reduce`, we do not need to do the reduce at this stage, and it will degrade the performance.